aptos-move/e2e-move-tests/tests/rotate_auth_key.rs

```diff
@@ -30,6 +32,50 @@ struct RotationProofChallenge {
     new_public_key: Vec<u8>,
 }
 
+#[test]
+fn rotate_auth_key_ed25519_to_ed25519() {
+    let mut harness = MoveHarness::new();
+    let mut account1 = harness.new_account_with_key_pair();
+
+    let account2 = harness.new_account_with_key_pair();
+    // assert that the payload is successfully processed (the signatures are correct)
+    assert_successful_payload_key_rotation_from_ed25519_to_ed25519(
+        &mut harness,
+        account1.clone(),
+        *account1.address(),
+        10,
+        account2.privkey.clone(),
+        account2.pubkey.clone(),
+    );
+    // rotate account1's keypair to account2
+    account1.rotate_key(account2.privkey, account2.pubkey);
+    // verify that we can still get to account1's originating address
+    verify_originating_address(&mut harness, account1.auth_key(), *account1.address());
+}
+
+#[test]
+fn rotate_auth_key_ed25519_to_multi_ed25519() {
```

Comment:
Same comment here: In the next PR,  can we coalesce this function and the previous one into one generic function?